sailthru_content/services/sailthru_translation_service.py

```diff
@@ -0,0 +1,275 @@
+import datetime
+import logging
+
+
+class SailthruTranslationService(object):
+    def __init__(self, data_service, lms_root, fixups=None):
+        self.data_service = data_service
+        self.lms_root = lms_root
+        self.fixups = fixups
+
+    def _convert_date(self, iso_date):
+        """ Convert date from ISO 8601 (e.g. 2016-04-15T20:35:11.424818Z) to Sailthru format
+        """
+        if iso_date is None:
+            return None
+        try:
+            if '.' in iso_date:
+                return datetime.datetime.strptime(iso_date, "%Y-%m-%dT%H:%M:%S.%fZ").strftime("%Y-%m-%d")
+            return datetime.datetime.strptime(iso_date, "%Y-%m-%dT%H:%M:%SZ").strftime("%Y-%m-%d")
+        except ValueError:
+            return None
+
+    def _convert_datetime(self, iso_date):
+        """ Convert date from ISO 8601 (e.g. 2016-04-15T20:35:11.424818Z) to Sailthru format
+        """
+        if iso_date is None:
+            return None
+        try:
+            if '.' in iso_date:
+                return datetime.datetime.strptime(iso_date, "%Y-%m-%dT%H:%M:%S.%fZ").strftime("%Y-%m-%d %H:%M:%S +0000")
+            return datetime.datetime.strptime(iso_date, "%Y-%m-%dT%H:%M:%SZ").strftime("%Y-%m-%d %H:%M:%S +0000")
+        except ValueError:
+            return None
+
+    def _convert_tag(self, tagtype, tag):
+        """ Convert string to a valid sailthru tag and add type- to the front if specified
+        """
+        if tag:
+            resp = tag.replace(' & ', '-').replace(',', '').replace('.', '').replace(' ', '-').replace('--', '-')
+            if tagtype:
+                resp = tagtype + '-' + resp
+            return resp
+        return ''
+
+    def _create_course_vars(self, course, course_run, url, site_name, program_dictionary=None):
+        """ Generate 'vars' section of Sailthru data
+        """
+        sailthru_content_vars = {}
+        sailthru_content_vars['course_run'] = True
+        sailthru_content_vars['marketing_url'] = url
+        sailthru_content_vars['course_id'] = course['key']
+        sailthru_content_vars['course_run_id'] = course_run['key']
+
+        # duplicate site_name in vars for purchase/enroll templates
+        if site_name:
+            sailthru_content_vars['site_name'] = site_name
+
+        # use enrollment_end for sailthru expire_date
+        if course_run.get('enrollment_end'):
+            sailthru_content_vars['enrollment_end'] = self._convert_date(course_run['enrollment_end'])
```

Comment:
Instead of a giant block of if statements, change the `_convert_date` method so that it accepts a dictionary and a key, and handles adding a new value only if the key is present in `course_run`. That way you write one if statement instead of six, and can use a for loop to handle the keys.